core/src/com/unciv/ui/options/GameplayTab.kt

```diff
@@ -35,4 +38,22 @@ fun gameplayTab(
     ) { settings.automatedWorkersReplaceImprovements = it }
     optionsPopup.addCheckbox(this, "Order trade offers by amount", settings.orderTradeOffersByAmount) { settings.orderTradeOffersByAmount = it }
     optionsPopup.addCheckbox(this, "Ask for confirmation when pressing next turn", settings.confirmNextTurn) { settings.confirmNextTurn = it }
+
+    addNotificationLogMaxTurnsSlider(this, settings, UncivGame.Current.worldScreen, optionsPopup.selectBoxMinWidth)
+}
+
+private fun addNotificationLogMaxTurnsSlider(table: Table, settings: GameSettings, screen: BaseScreen?, selectBoxMinWidth: Float) {
+    table.add("Notifications log max turns".toLabel()).left().fillX()
+
+    val minimapSlider = UncivSlider(
+        3f, 15f, 1f,
+        initial = settings.notificationsLogMaxTurns.toFloat()
+    ) {
+        val turns = it.toInt()
+        settings.notificationsLogMaxTurns = turns
+        settings.save()
+        if (screen is WorldScreen)
+            screen.shouldUpdate = true
```

Comment:
Doesn't this only affect the log that isn't in the worldscreen? Why update the worldscreen?